File: Lista nr 3/Circle.cpp
#include "Circle.hpp"
#define PI 3.14159265
#include "NegativeNumber.hpp"

Circle::Circle(int a)
{   
    if(a < 0)
        throw NegativeNumber();
    radius = a;
}

int Circle::area()
{
    return radius * radius * PI;
}
int Circle::perimeter()
{
    return 2 * radius * PI;
}
std::string Circle::getNAME()
{
    return "Circle";
}

Circle::~Circle()
{

}

File: Lista nr 3 Java/Circle.java
class Circle implements Figure
{
    private static final double PI = 3.14159265;
    private int radius;
    public Circle(int a) throws NegativeNumber
    {
        if(a < 0)
        throw new NegativeNumber();
        radius = a;
    }
    public int area()
    {
        return (int)(radius * radius * PI);
    }
    public int perimeter()
    {
        return (int)(2 * radius * PI);
    }
    public String getNAME()
    {
        return "Circle";
    }
}